File: introCS/java/CouponCollector.java
public class CouponCollector
{
   public static void main(String[] args)
   {
       int n = Integer.parseInt(args[0]);
       boolean[] isCollected = new boolean[n];
       int count = 0;
       int distinct = 0;
       
       while (distinct < n)
       {
           int r = (int) (Math.random() * n);
           count++;
           
           if (!isCollected[r])
           {
               distinct++;
               isCollected[r] = true;
           }
       }
       
       System.out.println(count);
   }
}

File: introCS/python/couponcollector.py
import random
import sys
import stdarray
import stdio

n = int(sys.argv[1])

count = 0
collectedCount = 0
isCollected = [False]*n

while collectedCount < n:
    value = random.randrange(0,n)
    count += 1
    if not isCollected[value]:
        collectedCount += 1
        isCollected[value] = True

print(count)